cli/types/index.d.ts

```diff
@@ -2458,6 +2463,11 @@ declare namespace Cypress {
     experimentalGetCookiesSameSite: boolean
   }
 
+  interface TestConfigOptions extends Partial<Pick<ConfigOptions, 'baseUrl' | 'defaultCommandTimeout' | 'animationDistanceThreshold' | 'waitForAnimations' | 'viewportHeight' | 'viewportWidth' | 'requestTimeout' | 'execTimeout' | 'env' | 'responseTimeout'>> {

```

Comment:
This is the whitelist of configuration values right? For docs? Plus `browser`? Does this also close this issue about changing `baseUrl` mid-test? https://github.com/cypress-io/cypress/issues/2918

Reply:
yes that is the whitelist. this does not fix the baseUrl issue, but makes fixing it possible

Reply:
So, will this tests pass below? (I tried to run this using this branch but it just throws `fn.apply is not a function`)

```js
it("visit example", { baseUrl: 'https://example.cypress.io'}, () => {
  cy.visit("/commands/querying")
  cy.url().should('eq', 'https://example.cypress.io/commands/querying')
})

it("visit docs", { baseUrl: 'https://docs.cypress.io' }, () => {
  cy.visit("/guides/overview/why-cypress.html")
  cy.url().should('eq', 'https://docs.cypress.io/guides/overview/why-cypress.html')
})
```

Reply:
added a test for this